test/unit/presenters/formats/service_sign_in_presenter_test.rb

```diff
@@ -0,0 +1,209 @@
+require 'test_helper'
+
+class ServiceSignInTest < ActiveSupport::TestCase
+  include GovukContentSchemaTestHelpers::TestUnit
+
+  def subject
+    Formats::ServiceSignInPresenter.new(@content)
+  end
+
+  def file_name
+    "example.yaml"
+  end
+
+  def load_content_from_file(file_name)
+    @content ||= YAML.load_file(Rails.root.join("lib", "service_sign_in", file_name)).deep_symbolize_keys
+  end
+
+  def setup
+    load_content_from_file(file_name)
+    @artefact ||= FactoryGirl.create(:artefact, kind: "transaction")
+    @parent ||= FactoryGirl.create(
+      :transaction_edition,
+      panopticon_id: @artefact.id,
+      slug: parent_slug
+    )
+  end
+
+  def parent_slug
+    "log-in-file-self-assessment-tax-return"
+  end
+
+  def base_path
+    "/#{parent_slug}/sign-in"
+  end
+
+  def result
+    subject.render_for_publishing_api
+  end
+
+  context "#content_id" do
+    should "create a new content id if we are creating a new content item" do
+      SecureRandom.stub :uuid, "random-uuid-string" do
+        publishing_api_has_lookups(base_path => nil)
+        assert_equal "random-uuid-string", subject.content_id
+      end
+    end
+
+    should "use existing content_id if content_item already exists in content-store" do
+      content_id = "random-content-id"
+      publishing_api_has_lookups(base_path => content_id)
+
+      assert_equal content_id, subject.content_id
```

Comment:
Here we're trying to test that the method returns the content_id for an existing content item instead of generating a new one for us, so it's  correct that we're checking `subject.content_id`.
Also `existing_content_id` is a private method so we won't be able to call that :)